packages/apollo-schemas/src/feature.schema.ts

```diff
@@ -1,22 +1,26 @@
 import { GFF3FeatureLineWithRefs } from '@gmod/gff'
-// import { Schema, SchemaFactory } from '@nestjs/mongoose'
-import { Document, Schema } from 'mongoose'
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose'
+import { Document, Schema as MongooseSchema } from 'mongoose'
 
-// export type FeatureDocument = Feature & Document
+import { RefSeq } from './refSeq.schema'
 
-// @Schema()
-// export class Feature {}
+export type FeatureDocument = Feature & Document
 
-// export const FeatureSchema = SchemaFactory.createForClass(Feature)
+@Schema()
+export class Feature {
+  @Prop({
+    required: true,
+    index: true,
+    type: MongooseSchema.Types.ObjectId,
+    ref: 'RefSeq',
+  })
+  refSeqId: RefSeq
 
-export const FeatureSchema = new Schema({
-  refSeqId: { type: String, required: true },
-  featureId: { type: [String], required: true },
-  gff3FeatureLineWithRefs: { type: JSON, required: true },
-})
-export interface FeatureDocument extends GFF3FeatureLineWithRefs, Document {
-  id: string
-  refSeqId: string
+  @Prop({ type: [String], required: true, index: true })
   featureId: string[]
+
+  @Prop({ type: JSON, required: true })
   gff3FeatureLineWithRefs: GFF3FeatureLineWithRefs
```

Comment:
I think we should spread the feature into the  top of the schema, for example like this:
```ts
@Schema()
export class Feature implements GFF3FeatureLineWithRefs {
  @Prop({
    required: true,
    index: true,
    type: MongooseSchema.Types.ObjectId,
    ref: 'RefSeq',
  })
  refSeqId: RefSeq

  @Prop({ type: [String], required: true, index: true })
  featureId: string[]

  @Prop({ required: true })
  // eslint-disable-next-line camelcase
  seq_id: string

  @Prop()
  source: string

  @Prop({ required: true })
  type: string

  @Prop({ required: true })
  start: number

  @Prop({ required: true })
  end: number

  @Prop()
  score: number

  @Prop()
  strand: string

  @Prop()
  phase: string

  @Prop({ type: Map, of: [String] })
  attributes: Record<string, string[]>

  @Prop({ type: JSON })
  // eslint-disable-next-line camelcase
  child_features: GFF3Feature[]

  @Prop({ type: JSON })
  // eslint-disable-next-line camelcase
  derived_features: GFF3Feature[]
}
```

Reply:
If we spread feature up to top level we have to think what shall we populate into those mandatory fields when we are inserting comments and directives into Features -collection in database.

Reply:
Moved to top level.